### DISCUSSION TITLE
what is difference between glsp-client and vscode-extension

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-glsp/glsp#1125] [General]
what is difference between glsp-client and vscode-extension,are they same?glsp-client and language-client also make me confused.my project need to intergate node-glsp-server,glsp-client,vscode-extension,vscode-webview,langium together

**Comment by planger** [comment]
I think the [Package Overview](https://eclipse.dev/glsp/documentation/integrations/) in the GLSP doc may help regarding this question. `glsp-client` is base framework for creating your specific diagram implementation (e.g. `my-glsp-client`). This package is platform independent and can thus be reused in any tool platform, such as VS Code or Eclipse Theia.

To integrate it into a tool platform, there is additional dedicated integration packages. The one for VS Code consists of a two parts:
* The `vscode-webview` packs the GLSP diagram into a single web page that is then integrated into a webview via an `iframe`
* The `vscode-extension` is the actual VS Code extension and includes the `vscode-webview` in order to register it via the VS Code extension API as a VS Code extension (see also [VS Code custom editor extensions](https://code.visualstudio.com/api/extension-guides/custom-editors))

**Comment by zrwsmd** [comment]
just use [glsp-vscode-integration-master ](https://github.com/eclipsesource/glsp-vscode-integration#workflow-diagram-example)project for example,which part is glsp-client

**Reply by planger** [reply]
I'm not sure I understand your question correctly. But the project you linked, depends on `workflow-glsp` (see [package.json](https://github.com/eclipsesource/glsp-vscode-integration/blob/master/example/workflow/webview/package.json#L30)), which in turn depends on `glsp-client`.

**Comment by zrwsmd** [comment]
i notice crossmodel project mainly intergate glsp with langium,while glsp-vscode-integration-master project intergate glsp with vscode-extension and vscode-webview,how could this two project combine together ,i mean intergate glsp with vscode-extension , vscode-webview ,langium together.but this two project seems incompatible with each other,for example,for launching glsp server code,their implement is total different like below code
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/de85ab48-fde8-4201-baf7-eed6ea5d1a49)

![image](https://github.com/eclipse-glsp/glsp/assets/26995040/d5324bc7-b99f-4f26-8764-86941adc9a86)

**Reply by planger** [reply]
The GLSP server is integrated with the Langium server in this case, whereas usually GLSP is a standalone process. Thus in this case the GLSP server is started as part of the process of the language server in Langium in order to gain access to the abstract syntax tree. The code to make this happen in langium is currently custom, but could eventually be generalized, I think.

**Comment by zrwsmd** [comment]
what is difference among part1 and part2 and GLSPStarter,is part1 for start glsp server and part2 for start glsp client?and what is the use of GLSPStarter.
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/f8a5bcf9-69b0-43ad-9879-1136058c32cb)
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/3bda96e7-9d96-40b2-a21a-2f607037544e)
